Wrappers/Python/test/test_out_in_place.py

```diff
@@ -313,3 +315,155 @@ def test_proximal_out(self):
+            AbsorptionTransmissionConverter(),
+            RingRemover(),
+            Slicer(roi={'horizontal':(None,None,2),'angle':(None,None,2)}), 
+            Binner(roi={'horizontal':(None,None,2),'angle':(None,None,2)}), 
+            Padder(pad_width=1)]
+        self.processor_out_same_size = [
+            True,
+            True,
+            True,
+            False,
+            False,
+            False
+        ]
+
+    def get_result(self, processor, data, *args):
+        input=data.copy() #To check that it isn't changed after function calls
+        try:
+            processor.set_input(data)
+            out = processor.get_output()
+            self.assertDataArraysInContainerAllClose(input, data, rtol=1e-5, msg= "In case processor.set_input(data), processor.get_output() where processor is  " + processor.__class__.__name__+ " the input data has been incorrectly affected by the calculation.")
+            return out
+        except NotImplementedError:
+            print("get_result test not implemented for  " + processor.__class__.__name__)
+            return None
+
+    def in_place_test(self, desired_result, processor, data, output_same_size=True):
+            if output_same_size==True:
+                out = data.copy()
+            else:
+                out=desired_result.copy()
+            try:
+                processor.set_input(data)
+                processor.get_output(out=out)
+                self.assertDataArraysInContainerAllClose(desired_result, out, rtol=1e-5, msg= "In place calculation failed for processor.set_input(data), processor.get_output(out=data) where processor is  " + processor.__class__.__name__+ "." )
+
+            except (InPlaceError, NotImplementedError):
+                print("in_place_test test not implemented for  " + processor.__class__.__name__)
+                pass
+
+    def out_test(self, desired_result, processor, data, output_same_size=True):
+        input = data.copy()
+        if output_same_size==True:
+            out=0*(data.copy())
+        else:
+            out=0*(desired_result.copy())
+
+        try:
+            processor.set_input(input)
+            output = processor.get_output(out=out)
+            
+            self.assertDataArraysInContainerAllClose(desired_result, out, rtol=1e-5, msg= "Calculation failed using processor.set_input(data), processor.get_output(out=out) where func is  " + processor.__class__.__name__+ ".")
+            self.assertDataArraysInContainerAllClose(input, data,  rtol=1e-5, msg= "In case processor.set_input(data), processor.get_output(out=data) where processor is  " + processor.__class__.__name__+ " the input data has been incorrectly affected by the calculation. ")
+            self.assertDataArraysInContainerAllClose(desired_result, output,  rtol=1e-5, msg= "In case processor.set_input(data), output=processor.get_output(out=data) where processor is  " + processor.__class__.__name__+ " the processor supresses the output. ")
```

Comment:
Does this always pass? Is out always returned? Just asking because this wasn't the case for a lot of the optimisation framework